modules/meta/arrow-meta-prototype/idea-plugin/src/main/kotlin/arrow/meta/dsl/ide/IdeSyntax.kt

```diff
@@ -1,6 +1,24 @@
 package arrow.meta.dsl.ide
 
-import arrow.meta.dsl.ide.feature1.Feature1Syntax
+import arrow.meta.dsl.ide.editor.action.AnActionSyntax
+import arrow.meta.dsl.ide.editor.color.ColorSyntax
+import arrow.meta.dsl.ide.editor.hints.HintingSyntax
+import arrow.meta.dsl.ide.editor.icon.IconProviderSyntax
+import arrow.meta.dsl.ide.editor.inspection.InspectionSyntax
+import arrow.meta.dsl.ide.editor.intention.IntentionExtensionProviderSyntax
+import arrow.meta.dsl.ide.editor.language.LanguageSyntax
+import arrow.meta.dsl.ide.editor.lineMarker.LineMarkerSyntax
+import arrow.meta.dsl.ide.editor.liveTemplate.LiveTemplateSyntax
+import arrow.meta.dsl.ide.editor.navigation.NavigationSyntax
+import arrow.meta.dsl.ide.editor.refactoring.RefactoringSyntax
+import arrow.meta.dsl.ide.editor.search.SearchSyntax
+import arrow.meta.dsl.ide.editor.structureView.StructureViewSyntax
+import arrow.meta.dsl.ide.editor.syntaxHighlighter.SyntaxHighlighterExtensionProviderSyntax
+import arrow.meta.dsl.ide.editor.usage.UsageSyntax
+import arrow.meta.dsl.ide.extensions.ExtensionProviderSyntax
 
-interface IdeSyntax : Feature1Syntax {
-}
\ No newline at end of file
+// TODO: Rewrite the Nullable in each ProductSyntax to proper Arrow Datatypes
```

Comment:
Most likely we will not use arrow in meta until we are able to shadow meta. Also what's wrong with nullables here?

Reply:
I just wanted clearer datatypes, e.g.: Option<IconProvider> rather than IconProvider?
But as you said, that would be in the long run.